content/lua/concepts/iterators/iterators.md

```diff
@@ -0,0 +1,77 @@
+---
+Title: 'Iterators' # Required; the file name should be the same as the title, but lowercase, with dashes instead of spaces, and all punctuation removed
+Description: 'Iterators are used to loop through a collection of data. They are also used to repeat processes a preset or infinite amount of times.' # Required; ideally under 150 characters and starts with a noun (used in search engine results and content previews)
+Subjects: # Please only use Subjects in the subjects.md file (https://github.com/Codecademy/docs/blob/main/documentation/subjects.md). If that list feels insufficient, feel free to create a new Subject and add it to subjects.md in your PR!
+  - 'Computer Science'
+  - 'Lua'
+Tags: # Please only use Tags in the tags.md file (https://github.com/Codecademy/docs/blob/main/documentation/tags.md). If that list feels insufficient, feel free to create a new Tag and add it to tags.md in your PR!
+  - 'Iterators'
+  - 'Conditionals'
+  - 'Tables'
+  - 'Arrays'
+  - 'Logical'
+CatalogContent: # Please use course/path landing page slugs, rather than linking to individual content items. If listing multiple items, please put the most relevant one first
+  - 'lua/iterators'
+---
+
+# **Iterators**
+
+**Iterators** are used to loop through a collection of data. They are also used to repeat processes a preset or infinite amount of times.
+
+## **Iteration Types**
+
+In Lua, there are multiple ways in which we can **iterate** through and/or repeat code with. Some iteration methods are intended for iterating through **Arrays**, while others can be used to
+**repeat** a command until/while a condition is met. Below are some examples of conditional iterations.
+
+```lua
+-- while CONDITION do
+-- # This iteration will repeatedly execute it's provided code until the CONDITION becomes false.
+x = 2
+
+while x == 2 do
```

Comment:
Issue: Not sure it's the greatest idea to put example code in our docs that create infinite loops. Maybe we put a `x = x-1` in the while loop and make the condition `x > 0`.